src/components/Upload/Forms/FormComponents/SubmitterInformation.js

```diff
@@ -2,43 +2,64 @@ import React, { Component } from 'react';
 
+const FIRST_NAME = "firstName";
+const LAST_NAME = "lastName";
+const EMAIL = "email";
+
 class SubmitterInformation extends Component {
-	
+
+	constructor(props) {
+		super(props);
+		this.state = this.getDerivedStateFromProps(props);
+	}
+
+    getDerivedStateFromProps(nextProps) {
+        return {
+            submitterFirstNameDisabled: this.isFieldDisabled(FIRST_NAME, nextProps),
+            submitterLastNameDisabled: this.isFieldDisabled(LAST_NAME, nextProps),
+            submitterEmailDisabled: this.isFieldDisabled(EMAIL, nextProps),
+            userInfoPopulated: this.isUserInfoPopulated(nextProps)
+        };
+    }
+
+	isFieldDisabled(fieldName, props) {
+        return props.userInformation.hasOwnProperty(fieldName)
+            && props.userInformation[fieldName] !== "";
+	}
+
+	isUserInfoPopulated(props) {
+		return this.isFieldDisabled(FIRST_NAME, props)
+            && this.isFieldDisabled(LAST_NAME, props)
+            && this.isFieldDisabled(EMAIL, props)
+	}
+
 	render() {
-		
-		let submitterFirstNameDisabled = this.props.userInformation.firstName !== "";
-		let submitterLastNameDisabled = this.props.userInformation.lastName !== "";
-		let submitterEmailDisabled = this.props.userInformation.email !== "";
-		
-		let dontNeedUserInfo = submitterFirstNameDisabled && submitterLastNameDisabled && submitterEmailDisabled;
-		
-		if (dontNeedUserInfo) {
-			return (
-				<Col sm={12} md={12} lg={12} className="ant-form-item submitterInfo">
-					<div className="ant-form-item-label">
-						<label>Submitted By</label>
-					</div>
-					<div className="submitterInfoValues">
-						{this.props.userInformation.firstName} {this.props.userInformation.lastName} ({this.props.userInformation.email})
-					</div>
-				</Col>
-			);
-		} else {
-			return(
-	    		 <div className="row w-100 ml-0">
-					<Col sm={12} md={6} lg={4}>
-						<TextField label="First Name" fieldName="submitterFirstName" isDisabled={submitterLastNameDisabled} form={this.props.form} isRequired={true}/>
-					</Col>
-					<Col sm={12} md={6} lg={4} >
-						<TextField label="Last Name" fieldName="submitterLastName" isDisabled={submitterLastNameDisabled} form={this.props.form} isRequired={true}/>
-					</Col>
-					<Col sm={12} md={6} lg={4}>
-						<TextField label="Email" fieldName="submitterEmail" form={this.props.form} isRequired={true}/>
-					</Col>
-				</div>
-			);
-		}
+        if (this.state.userInfoPopulated) {
+            return (
+                <Col sm={12} md={12} lg={12} className="ant-form-item submitterInfo">
+                    <div className="ant-form-item-label">
+                        <label>Submitted By</label>
+                    </div>
+                    <div className="submitterInfoValues">
+                        {this.props.userInformation.firstName} {this.props.userInformation.lastName} ({this.props.userInformation.email})
+                    </div>
+                </Col>
+            );
+        }
 
+		return (
```

Comment:
I would really love to see this in an else.  I know that programatically it makes no difference, but to a person skimming the code it makes a big difference.